Build and install my own dependencies in a dockcross build

Building for linux-armv7, I see the following [lines in the toolchain](https://github.com/dockcross/dockcross/blob/master/linux-armv7/Toolchain.cmake#L13-L16):

```cmake
set(CMAKE_FIND_ROOT_PATH ${cross_root} ${cross_root}/${cross_triple})
set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
```

Which means that `find_*` will look for dependencies in `/usr/xcc` and `/usr/xcc/armv7-unknown-linux-gnueabi`, right?

Now say I build my own dependencies, that I can either install locally and add to `CMAKE_PREFIX_PATH`, or install somewhere on the system. Because of `ONLY` (above), the `CMAKE_PREFIX_PATH` strategy will not work (it will be ignored).

So I tried to install my dependencies in `/usr/xcc`. And that doesn't work either, because of some permission error:

```
CMake Error at cmake_install.cmake:41 (file):
  file cannot create directory: /usr/xcc/lib.  Maybe need administrative
  privileges
```

What would be the recommended way to deal with that? And is there a reason why I could not install my dependencies there? I don't really get where the permission comes from :confused:.

--- a/linux-armv7/Toolchain.cmake
+++ b/linux-armv7/Toolchain.cmake
@@ -12,8 +12,8 @@ set(CMAKE_CXX_FLAGS "-I ${cross_root}/include/")
 
 set(CMAKE_FIND_ROOT_PATH ${cross_root} ${cross_root}/${cross_triple})
 set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
-set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH)
 set(CMAKE_SYSROOT ${cross_root}/${cross_triple}/sysroot)
 
 set(CMAKE_CROSSCOMPILING_EMULATOR /usr/bin/qemu-arm)

